compute_endpoint/tests/unit/test_endpoint_unit.py

```diff
@@ -158,178 +109,133 @@ def _wrapped_umask(new_umask: int | None) -> int:
 
 
 @pytest.fixture
-def mock_reg_info():
+def uname(randomstring):
+    return randomstring()
+
+
+@pytest.fixture
+def pword(randomstring):
+    return randomstring()
+
+
+@pytest.fixture
+def mock_reg_info(ep_uuid, uname, pword):
+    c_url = f"amqp://{uname}:{pword}@some.domain"
     yield {
-        "endpoint_id": str(uuid.uuid4()),
-        "task_queue_info": {"connection_url": "amqp://some.domain:1234"},
-        "result_queue_info": {"connection_url": "amqp://some.domain"},
-        "heartbeat_queue_info": {"connection_url": "amqp://some.domain"},
+        "endpoint_id": ep_uuid,
+        "task_queue_info": {"connection_url": f"{c_url}:1234"},
+        "result_queue_info": {"connection_url": c_url},
+        "heartbeat_queue_info": {"connection_url": c_url},
     }
 
 
-@responses.activate
-def test_start_endpoint(
-    mocker,
-    fs,
-    randomstring,
-    get_standard_compute_client,
-    register_endpoint_response,
-    mock_ep_data,
+def test_start_endpoint_no_reg_provided_registers(
+    mock_daemon, mock_log, mock_gcc, mock_get_client, mock_ep_data, ep_uuid
 ):
-    mock_gcc = get_standard_compute_client()
-    mock_log = mocker.patch(f"{_mock_base}log")
-    mock_daemon = mocker.patch(f"{_mock_base}daemon")
-    mock_epinterchange = mocker.patch(f"{_mock_base}EndpointInterchange")
-    mocker.patch(f"{_mock_base}Endpoint.get_funcx_client").return_value = mock_gcc
-
     ep, ep_dir, log_to_console, no_color, ep_conf = mock_ep_data
-    ep_id = str(uuid.uuid4())
+    ep_args = (ep_dir, ep_uuid, ep_conf, log_to_console, no_color)
 
-    uname, pword = randomstring(), randomstring()
-    register_endpoint_response(endpoint_id=ep_id, username=uname, password=pword)
+    ep_json_p = ep_dir / "endpoint.json"
+    assert not ep_json_p.exists(), "Verify test setup"
 
-    ep.start_endpoint(
-        ep_dir, ep_id, ep_conf, log_to_console, no_color, reg_info={}, ep_info={}
-    )
+    with mock.patch(f"{_mock_base}Endpoint.daemon_launch") as mock_launch:
```

Comment:
``with mock.patch(f"{_mock_base}Endpoint.daemon_launch") as mock_launch:``
is used 4 times in this PR, since you are adding a few fixtures, why not add that?